### PR TITLE
Quick Sort Implemented

### PR DESCRIPTION
**Pull request by SamSaver**
Samarth Singh Pawar
11941060
samarths@iitbhilai.ac.in

### CODE REVIEW

File: Sorting/quick.cpp [hunk 1]
```diff
@@ -7,6 +7,35 @@
  *
  * Returns: null
  */
-void Sort::QuickSort(){
-    std::cout<<"[TODO @ Sorting/quick.cpp] QuickSort() needs to be implemented.\n";
+
+ //QUICK SORT ALGORITHM
```
Comment (yatendra1999) [inline Sorting/quick.cpp]: Unnecessary comment. Documentation will take care. Only add comment if very crucial design decision is there which others should keep in mind.

Comment (SamSaver) [reply Sorting/quick.cpp]: @yatendra1999 Done.

File: Sorting/quick.cpp [hunk 2]
```diff
@@ -7,6 +7,33 @@
  *
  * Returns: null
  */
-void Sort::QuickSort(){
-    std::cout<<"[TODO @ Sorting/quick.cpp] QuickSort() needs to be implemented.\n";
+ void Swap(int &a,int &b){
+   int t=a;
+   a=b;
+   b=t;
+}
```
Comment (shashwatj07) [inline Sorting/quick.cpp]: put this in sorting.cpp as it is used by many algorithms. You can also use std::swap(&int, &int)

File: Sorting/sorting.hpp [hunk 3]
```diff
@@ -20,7 +20,7 @@ class Sort {
     void BubbleSort();
     void InsertionSort();
     void MergeSort();
-    void QuickSort();
+    void QuickSort(int p=0, int r=-10);
```
Comment (shashwatj07) [inline Sorting/sorting.hpp]: ```int r=-10```
Why is hardcoding required? It's a bad practice. Is there some other way we can handle this?

**Review by shashwatj07 — CHANGES_REQUESTED** [review]

**Comment by shashwatj07** [comment]
@yatendra1999 ping

File: Sorting/quick.cpp (hunk 2 above)
Comment (SamSaver) [reply Sorting/quick.cpp]: To use std::swap(&int, &int), I need to add <algorithm> library in quick.cpp. Am I allowed to do that?

Comment (shashwatj07) [reply Sorting/quick.cpp]: I think it'll be better to add the function to sorting.cpp
Including a library is also fine. It'll just increase the compile time a bit.

Comment (SamSaver) [reply Sorting/quick.cpp]: @shashwatj07 Done.

File: Sorting/sorting.hpp (hunk 3 above)
Comment (SamSaver) [reply Sorting/sorting.hpp]: @shashwatj07 Okay done.

**Review by yatendra1999 — APPROVED** [review]
LGTM

**Comment by shashwatj07** [comment]
rebase and resolve conflict @SamSaver

**Comment by SamSaver** [comment]
> rebase and resolve conflict @SamSaver

@shashwatj07 Resolved.

**Review by shashwatj07 — APPROVED** [review]

**Comment by shashwatj07** [comment]
ready to merge

**Pull request merged**